File: leakcanary-analyzer/src/main/java/com/squareup/leakcanary/Exclusion.java
package com.squareup.leakcanary;

import java.io.Serializable;

public final class Exclusion implements Serializable {
  public final String name;
  public final String reason;
  public final boolean alwaysExclude;
  public final String matching;

  Exclusion(ExcludedRefs.ParamsBuilder builder) {
    this.name = builder.getName();
    this.reason = builder.getReason();
    this.alwaysExclude = builder.getAlwaysExclude();
    this.matching = builder.getMatching();
  }
}

File: leakcanary-analyzer/src/main/java/com/squareup/leakcanary/Exclusion.kt
package com.squareup.leakcanary

import java.io.Serializable

class Exclusion internal constructor(builder: ExcludedRefs.ParamsBuilder) : Serializable {
  val name: String? = builder.name
  val reason: String? = builder.reason
  val alwaysExclude: Boolean = builder.alwaysExclude
  val matching: String = builder.matching
}
